docs/guides/modals.md

```diff
@@ -0,0 +1,409 @@
+
+### Slack App Configuration
+
+The first step to use modals in your app is to enable Interactive Components. Visit the [Slack App configuration page](http://api.slack.com/apps), choose the app you're working on, and go to **Features** > **Interactive Components** on the left pain. There are three things to do on the page.
+
+* Turn on the feature
+* Set the **Request URL** to `https://{your app's public URL domain}/slack/events`
+* Click the **Save Changes** button at the bottom for sure
+
+### What Your Bolt App Does
+
+There are three patterns to handle on modals. As always, your app has to respond to the request within 3 seconds by `ack()` method for sure. Otherwise, the user will see the timeout error on Slack.
+
+#### block_actions requests
+
+When someone uses an [interactive component](https://api.slack.com/reference/block-kit/interactive-components) in your app's modal views, the app will receive a [block_actions payload](https://api.slack.com/reference/interaction-payloads/block-actions). All you need to do to handle Slack requests by user interactions are:
+
+1. Verify requests from Slack (read [this](https://api.slack.com/docs/verifying-requests-from-slack) if unfamiliar)
+1. Parse the request body, and check if the `action_id` in a block is the one you'd like to handle
+1. [Modify/push a view via API](https://api.slack.com/surfaces/modals/using#modifying) and/or update the modal to hold the sent data as [private_metadata](https://api.slack.com/surfaces/modals/using#carrying_data_between_views)
+1. Respond with 200 OK as the acknowledgment
+
+#### view_submission requests
+
+When a modal view is submitted, you'll receive a [view_submission payload](https://api.slack.com/reference/interaction-payloads/views#view_submission). All you need to do to handle Slack requests by user interactions are:
+
+1. Verify requests from Slack (read [this](https://api.slack.com/docs/verifying-requests-from-slack) if unfamiliar)
+1. Parse the request body, and check if the `type` is `view_submission` and the `callback_id` is the one you'd like to handle
+1. Extract the form data from `view.state.values`
+1. Do whatever to do such as input validations, storing them in a database, talking to external services
+1. Respond with 200 OK as the acknowledgment by either of the followings:
+  * Sending an empty body means closing only the modal
+  * Sending a body with `response_action` (possible values are `errors`, `update`, `push`, `clear`)
+
+#### view_closed requests (only when `notify_on_close` is true)
+
+Your app can optionally receive [view_closed payloads](https://api.slack.com/reference/interaction-payloads/views#view_closed) whenever a user clicks on the Cancel or x buttons. These buttons are standard in all app modals. All you need to do to handle Slack requests by user interactions are:
```

Comment:
It'd be helpful to note how to receive `view_closed` payloads (talked about in https://api.slack.com/surfaces/modals/using)